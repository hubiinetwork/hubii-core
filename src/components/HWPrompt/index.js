/**
*
* HWPrompt
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import { getAbsolutePath } from 'utils/electron';

import Text from 'components/ui/Text';

import {
  OuterWrapper,
  Row,
  KeyText,
  DescriptiveIcon,
  StatusIcon,
  SingleRowWrapper,
  SingleRowIcon,
  ConfTxDeviceImg,
  ConfTxShield,
} from './style';

const confTxOnDevicePrompt = (deviceType) => (
  <SingleRowWrapper>
    <ConfTxShield src={getAbsolutePath('public/images/shield.png')} />
    <ConfTxDeviceImg src={getAbsolutePath(`public/images/conf-tx-${deviceType}.png`)} />
    <Text large>{ 'Verify the details shown on your device' }</Text>
  </SingleRowWrapper>
);

const singleRowMsg = (msg, iconType, iconColor) => (
  <SingleRowWrapper>
    <SingleRowIcon type={iconType || 'loading'} color={iconColor} />
    <Text large>{msg}</Text>
  </SingleRowWrapper>
);

const lnsPrompt = (ledgerInfo) => {
  const error = ledgerInfo.get('error');
  // check if device is connected
  if (ledgerInfo.get('ethConnected')) {
    return singleRowMsg(
        'Device connection established',
        'check'
      );
  }

  // check the error
  if (error && error.includes("Ledger connected but does not appear to have 'Browser support'")) {
    return singleRowMsg(
      "Please set 'Browser support' to 'No' in your device settings to continue"
    );
  } else if
  (
    error
    && !error.includes('does not appear to have the Ethereum app open')
    && !error.includes('Ledger could not be detected')
  ) {
    return singleRowMsg(
      'Something went wrong, please reconnect your device and try again',
      'exclamation-circle-o',
      'orange'
    );
  }
  // no errors, show prompts
  let stage;
  if (!ledgerInfo.get('connected')) {
    stage = 'connect';
  } else if (!ledgerInfo.get('ethConnected')) {
    stage = 'openApp';
  } else {
    return singleRowMsg(
        'Device connection established',
        'check'
    );
  }
  return (
    <OuterWrapper>
      <Row pos="top" active={stage === 'connect'}>
        <DescriptiveIcon
          src={stage === 'connect' ?
            getAbsolutePath('public/images/hw-wallet-usb-green.png') :
            getAbsolutePath('public/images/hw-wallet-usb-white.png')
          }
        />
        <Text>Connect and unlock your <KeyText>Ledger Device</KeyText></Text>
        <StatusIcon
          type={stage === 'connect' ? 'loading' : 'check'}
        />
      </Row>
      <Row pos="bottom" active={stage === 'openApp'}>
        <DescriptiveIcon
          src={stage === 'connect' ?
            getAbsolutePath('public/images/hw-wallet-eth-grey.png') :
            getAbsolutePath('public/images/hw-wallet-eth-green.png')
          }
        />
        <Text>Open the <KeyText>Ethereum</KeyText> app on your device</Text>
        <StatusIcon
          type={stage === 'openApp' ? 'loading' : 'ellipsis'}
        />
      </Row>
    </OuterWrapper>
  );
};

const trezorPrompt = (trezorInfo) => {
  const error = trezorInfo.get('error');
  if (error && !error.includes('Trezor is not connected')) {
    return singleRowMsg(
        'Something went wrong, please reconnect your device and try again',
        'exclamation-circle-o',
        'orange'
      );
  }

    // device connection is good
  if (trezorInfo.get('status') === 'connected') {
    return singleRowMsg(
        'Device connection established',
        'check'
      );
  }

    // trezor needs to be connected
  return singleRowMsg(
      'Please connect and unlock your Trezor device to continue'
    );
};

class HWPrompt extends React.Component { // eslint-disable-line react/prefer-stateless-function
  render() {
    const {
      deviceType,
      ledgerInfo,
      trezorInfo,
    } = this.props;

    const confTxOnDevice = deviceType === 'lns'
      ? ledgerInfo.get('confTxOnDevice')
      : trezorInfo.get('confTxOnDevice');

    // tx needs to be confirmed on device
    if (confTxOnDevice) {
      return confTxOnDevicePrompt(deviceType);
    }

    // handle ledger
    if (deviceType === 'lns') {
      return lnsPrompt(ledgerInfo);
    }

    // handle trezor
    return trezorPrompt(trezorInfo);
  }
}

HWPrompt.propTypes = {
  deviceType: PropTypes.oneOf(['lns', 'trezor']),
  ledgerInfo: PropTypes.object.isRequired,
  trezorInfo: PropTypes.object.isRequired,
};

export default HWPrompt;
