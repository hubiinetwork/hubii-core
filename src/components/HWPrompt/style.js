import styled from 'styled-components';
import { Icon } from 'antd';

export const OuterWrapper = styled.div`
  display: flex;
  width: 33rem;
  flex-direction: column;
`;

export const DescriptiveIcon = styled(Icon)`
  color: ${(props) => props.active === 'true' ? props.theme.palette.info : null};
  font-size: 1.5rem;
  margin-right: 1rem;
`;

export const SingleRowWrapper = styled.div`
  display: flex;
  align-items: center;
  /* justify-content: space-between; */
  width: 33rem;
  border: 0.15rem solid ${({ theme }) => theme.palette.secondary5};
  border-radius: 0.5rem; 
  color: white;
  font-size: 1.25rem;
  padding: 2rem 2rem;
`;

export const SingleRowIcon = styled(Icon)`
  font-size: 3rem;
  margin-right: 2rem;
  color: ${(props) => props.color || props.theme.palette.info};
`;

export const StatusIcon = styled(Icon)`
  font-size: 1.5rem;
  margin-left: auto;
  color: ${(props) => props.type === 'check' ? props.theme.palette.info : null};
`;

export const P = styled.p`
  margin-bottom: 0;
`;

export const Row = styled.div`
  padding: 2rem;
  display: flex;
  align-items: center;
  width: 100%;
  border: ${(props) => props.active ?
    `0.3rem solid ${props.theme.palette.info};` :
    `0.15rem solid ${props.theme.palette.secondary5};`
  };
  color: ${(props) => props.active ? 'white' : `${props.theme.palette.secondary5}`};
  border-radius: ${(props) => {
    if (props.pos === 'top') {
      return '0.5rem 0.5rem 0 0';
    }
    return '0 0 0.5rem 0.5rem';
  }};
  ${(props) => {
    if (!props.active && props.pos === 'top') {
      return `
        border-bottom: 0;
        color: white;
      `;
    }
    if (!props.active && props.pos === 'bottom') {
      return 'border-top: 0';
    }
    return null;
  }};
`;

export const KeyText = styled.span`
  font-weight: bold;
`;
