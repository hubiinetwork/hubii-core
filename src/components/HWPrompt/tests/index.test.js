import React from 'react';
import { shallow } from 'enzyme';

import HWPrompt from '../index';

describe('<HWPrompt />', () => {
  const props = {
    deviceType: 'lns',
    error: 'Ledger could not be detected',
    confTxOnDevice: false,
  };
  it('should render correctly in lns connect stage', () => {
    const wrapper = shallow(<HWPrompt {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly in lns openApp stage', () => {
    const wrapper = shallow(
      <HWPrompt
        {...props}
        error="some unknown error~~~~~~~~"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly in trezor connect stage', () => {
    const wrapper = shallow(
      <HWPrompt
        {...props}
        deviceType="trezor"
      />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when device is connected with no errors', () => {
    const wrapper = shallow(
      <HWPrompt
        {...props}
        error={null}
      />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when tx conf required on LNS', () => {
    const wrapper = shallow(
      <HWPrompt
        {...props}
        deviceType="lns"
        confTxOnDevice
      />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when tx conf required on trezor', () => {
    const wrapper = shallow(
      <HWPrompt
        {...props}
        deviceType="trezor"
        confTxOnDevice
      />);
    expect(wrapper).toMatchSnapshot();
  });
});
