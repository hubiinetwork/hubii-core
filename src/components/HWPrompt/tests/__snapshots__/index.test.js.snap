// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HWPrompt /> should render correctly in lns connect stage 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HWPrompt
    deviceType="lns"
    error="Ledger could not be detected"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <style__Row
          active={true}
          pos="top"
>
          <style__DescriptiveIcon
                    src="/public/images/hw-wallet-usb-green.png"
          />
          <style__P>
                    Connect and unlock your 
                    <style__KeyText>
                              Ledger Device
                    </style__KeyText>
          </style__P>
          <style__StatusIcon
                    type="loading"
          />
</style__Row>,
        <style__Row
          active={false}
          pos="bottom"
>
          <style__DescriptiveIcon
                    src="/public/images/hw-wallet-eth-grey.png"
          />
          <style__P>
                    Open the 
                    <style__KeyText>
                              Ethereum
                    </style__KeyText>
                     app on your device
          </style__P>
          <style__StatusIcon
                    type="ellipsis"
          />
</style__Row>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": true,
          "children": Array [
            <style__DescriptiveIcon
              src="/public/images/hw-wallet-usb-green.png"
/>,
            <style__P>
              Connect and unlock your 
              <style__KeyText>
                            Ledger Device
              </style__KeyText>
</style__P>,
            <style__StatusIcon
              type="loading"
/>,
          ],
          "pos": "top",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "src": "/public/images/hw-wallet-usb-green.png",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "Connect and unlock your ",
                <style__KeyText>
                  Ledger Device
</style__KeyText>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Connect and unlock your ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Ledger Device",
                },
                "ref": null,
                "rendered": "Ledger Device",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "type": "loading",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": false,
          "children": Array [
            <style__DescriptiveIcon
              src="/public/images/hw-wallet-eth-grey.png"
/>,
            <style__P>
              Open the 
              <style__KeyText>
                            Ethereum
              </style__KeyText>
               app on your device
</style__P>,
            <style__StatusIcon
              type="ellipsis"
/>,
          ],
          "pos": "bottom",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "src": "/public/images/hw-wallet-eth-grey.png",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "Open the ",
                <style__KeyText>
                  Ethereum
</style__KeyText>,
                " app on your device",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Open the ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Ethereum",
                },
                "ref": null,
                "rendered": "Ethereum",
                "type": [Function],
              },
              " app on your device",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "type": "ellipsis",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <style__Row
            active={true}
            pos="top"
>
            <style__DescriptiveIcon
                        src="/public/images/hw-wallet-usb-green.png"
            />
            <style__P>
                        Connect and unlock your 
                        <style__KeyText>
                                    Ledger Device
                        </style__KeyText>
            </style__P>
            <style__StatusIcon
                        type="loading"
            />
</style__Row>,
          <style__Row
            active={false}
            pos="bottom"
>
            <style__DescriptiveIcon
                        src="/public/images/hw-wallet-eth-grey.png"
            />
            <style__P>
                        Open the 
                        <style__KeyText>
                                    Ethereum
                        </style__KeyText>
                         app on your device
            </style__P>
            <style__StatusIcon
                        type="ellipsis"
            />
</style__Row>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": true,
            "children": Array [
              <style__DescriptiveIcon
                src="/public/images/hw-wallet-usb-green.png"
/>,
              <style__P>
                Connect and unlock your 
                <style__KeyText>
                                Ledger Device
                </style__KeyText>
</style__P>,
              <style__StatusIcon
                type="loading"
/>,
            ],
            "pos": "top",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "src": "/public/images/hw-wallet-usb-green.png",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  "Connect and unlock your ",
                  <style__KeyText>
                    Ledger Device
</style__KeyText>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Connect and unlock your ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Ledger Device",
                  },
                  "ref": null,
                  "rendered": "Ledger Device",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "type": "loading",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "children": Array [
              <style__DescriptiveIcon
                src="/public/images/hw-wallet-eth-grey.png"
/>,
              <style__P>
                Open the 
                <style__KeyText>
                                Ethereum
                </style__KeyText>
                 app on your device
</style__P>,
              <style__StatusIcon
                type="ellipsis"
/>,
            ],
            "pos": "bottom",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "src": "/public/images/hw-wallet-eth-grey.png",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  "Open the ",
                  <style__KeyText>
                    Ethereum
</style__KeyText>,
                  " app on your device",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Open the ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Ethereum",
                  },
                  "ref": null,
                  "rendered": "Ethereum",
                  "type": [Function],
                },
                " app on your device",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "type": "ellipsis",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<HWPrompt /> should render correctly in lns openApp stage 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HWPrompt
    deviceType="lns"
    error="some unknown error~~~~~~~~"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <style__SingleRowIcon
          color="orange"
          type="exclamation-circle-o"
/>,
        <style__P>
          Something went wrong, please reconnect your device and try again
</style__P>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "color": "orange",
          "type": "exclamation-circle-o",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Something went wrong, please reconnect your device and try again",
        },
        "ref": null,
        "rendered": "Something went wrong, please reconnect your device and try again",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <style__SingleRowIcon
            color="orange"
            type="exclamation-circle-o"
/>,
          <style__P>
            Something went wrong, please reconnect your device and try again
</style__P>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "color": "orange",
            "type": "exclamation-circle-o",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Something went wrong, please reconnect your device and try again",
          },
          "ref": null,
          "rendered": "Something went wrong, please reconnect your device and try again",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<HWPrompt /> should render correctly in trezor connect stage 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HWPrompt
    deviceType="trezor"
    error="Ledger could not be detected"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <style__SingleRowIcon
          color={undefined}
          type="loading"
/>,
        <style__P>
          Please connect and unlock your Trezor device to continue
</style__P>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "color": undefined,
          "type": "loading",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Please connect and unlock your Trezor device to continue",
        },
        "ref": null,
        "rendered": "Please connect and unlock your Trezor device to continue",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <style__SingleRowIcon
            color={undefined}
            type="loading"
/>,
          <style__P>
            Please connect and unlock your Trezor device to continue
</style__P>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "color": undefined,
            "type": "loading",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Please connect and unlock your Trezor device to continue",
          },
          "ref": null,
          "rendered": "Please connect and unlock your Trezor device to continue",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<HWPrompt /> should render correctly when device is connected with no errors 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HWPrompt
    deviceType="lns"
    error={null}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <style__SingleRowIcon
          color={undefined}
          type="check"
/>,
        <style__P>
          Device connection established
</style__P>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "color": undefined,
          "type": "check",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Device connection established",
        },
        "ref": null,
        "rendered": "Device connection established",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <style__SingleRowIcon
            color={undefined}
            type="check"
/>,
          <style__P>
            Device connection established
</style__P>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "color": undefined,
            "type": "check",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Device connection established",
          },
          "ref": null,
          "rendered": "Device connection established",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
